source_files/edge/rad_act.cc

```diff
@@ -903,6 +903,30 @@ void RAD_ActLightSector(rad_trigger_t *R, void *param)
 	}
 }
 
+void RAD_ActChangeSectorType(rad_trigger_t *R, void *param)
+{
+	s_sectortypechanger_t *t = (s_sectortypechanger_t *) param;
+	int i;
+
+	for (i=0; i < numsectors; i++)
+	{
+		if (sectors[i].tag != t->tag) continue;
+
+		P_SectorChangeSpecial(sectors + i, t->typenum);
+
+		if (sectors[i].props.special && sectors[i].props.special->fog_color != RGB_NO_VALUE)
+		{
+			sectors[i].props.fog_color = sectors[i].props.special->fog_color;
+			sectors[i].props.fog_density = 0.01f * sectors[i].props.special->fog_density;
```

Comment:
When changing fog properties for a sector, it is also important to invalidate any existing fogwalls (the translucent "faux walls" that reflect the fog color and density of the adjacent sector). See the last portion of the RAD_ActFogSector function for a reference.

Reply:
Fixed. Added a few comments to RAD_ActChangeSectorType too.